server/routerlicious/packages/services-utils/src/redisClientConnectionManager.ts

```diff
@@ -129,4 +163,33 @@ export class RedisClientConnectionManager implements IRedisClientConnectionManag
 		}
 		return this.client;
 	}
+
+	public addErrorHandler(
+		lumberProperties: Record<string, any> = {},
+		errorMessage: string = "Error with Redis",
+		additionalLoggingFunctionality?: (error: Error) => boolean,
+	): void {
+		if (!this.client) {
+			throw new Error("Redis client not initialized");
+		}
+
+		this.client.on("error", (error) => {
```

Comment:
Check with @dhr-verma about this, but I think this needs to be changed (maybe). IIRC the internal client for the connection manager can be invalidated or recreated for auth purposes. There are even some tests that do this. If it's recreated, this listener will not be present on the new client instance.

Maybe you could add a private EventEmitter with this listener registered, and then add event forwarding (like on("event", () => this.emitter.emit("event")). Not sure if necessary or overkill. Check with Dhruv

Reply:
This also might be an existing issue or outdated information. This is probably equivalent logic. Just worth checking

Reply:
If this is the case, then I think it is an existing issue, because the addErrorHandler is pretty much the equivalent logic. The function replaces any calls of this.client.on("error" in the code, and does mostly the same thing, with a few additional pieces of logging logic.

Will check with Dhruv though to see if this is something that should be changed

Reply:
I think its also possible that this is something for FRS's AzureRedisClientConnectionManager

Reply:
Talked with Dhruv about this. The issue you mentioned is only present on the FRS side due to some socket.io issues, the OSS side no longer has this client invalidation behavior, so it is good to stay as-is here. 
On the FRS side, the logs will have to be added a bit differently.